### PR TITLE
Java skeleton

### PR DESCRIPTION
**Pull request by lsavouillannxw**

### CODE REVIEW

File: templates/java/skeleton/templates/application.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  context-path: ${APPLICATION_CONTEXT_PATH:/}
+  port: ${APPLICATION_PORT:8085}
+  compression:
+    enabled: ${APPLICATION_COMPRESSION_ENABLED:true}
+    mime-types: ${APPLICATION_MINE_TYPES:application/json}
+    min-response-size: ${APPLICATION_MIN_RESPONSE_SIZE:1024}
+
+#management:
+#  security:
+#    enabled: ${APPLICATION_SECURITY_ENABLED:true} # set to false to disable 'default' Spring Boot security
```
Comment (cpage-nexway) [inline templates/java/skeleton/templates/application.yml]: This allow to have by spring configuration a status endpoint (or ping for example):
`endpoints:
  health:
    id: health
    path: status`

However, I didn't find how to modify easily the payload.
Thoughts ?

File: templates/java/skeleton/templates/ApiQueries.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	/* GET : ping test */
+    @ApiOperation(value="ping", notes="ping test")
+    @RequestMapping(value = "/ping", method = RequestMethod.GET, produces = "application/json")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Success", response = Void.class),
+            @ApiResponse(code = 401, message = "Unauthorized"),
+            @ApiResponse(code = 403, message = "Forbidden"),
+            @ApiResponse(code = 404, message = "Not Found"),
+            @ApiResponse(code = 500, message = "Failure")})
+    public String ping() {
```
Comment (cpage-nexway) [inline templates/java/skeleton/templates/ApiQueries.java]: imo, it's better to use standard springboot endpoint. See application.yml comment.

File: templates/java/skeleton/templates/ApiQueries.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+	/* GET : retrieved all resource */
+    @ApiOperation(value="getAll", notes="retrieved all resource")
+    @RequestMapping(method = RequestMethod.GET, produces = "application/json")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Success", response = Void.class),
+            @ApiResponse(code = 401, message = "Unauthorized"),
+            @ApiResponse(code = 403, message = "Forbidden"),
+            @ApiResponse(code = 404, message = "Not Found"),
+            @ApiResponse(code = 500, message = "Failure")})
+    public void getAll() {
```
Comment (cpage-nexway) [inline templates/java/skeleton/templates/ApiQueries.java]: imo, getall has always to accept a query param which is a list of ids.
Thoughts ?

File: templates/java/skeleton/templates/Application.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import springfox.documentation.builders.ApiInfoBuilder;
+import springfox.documentation.builders.RequestHandlerSelectors;
+import springfox.documentation.service.ApiInfo;
+import springfox.documentation.spi.DocumentationType;
+import springfox.documentation.spring.web.plugins.Docket;
+import springfox.documentation.swagger2.annotations.EnableSwagger2;
+
+@SpringBootApplication
```
Comment (cpage-nexway) [inline templates/java/skeleton/templates/Application.java]: http://docs.spring.io/autorepo/docs/spring-boot/current/reference/html/using-boot-using-springbootapplication-annotation.html

It includes @EnableAutoConfiguration and @ComponentScan.

File: templates/java/skeleton/templates/application.yml (hunk 1 above)
Comment (lsavouillannxw) [inline templates/java/skeleton/templates/application.yml]: http://docs.spring.io/spring-boot/docs/current/reference/html/production-ready-endpoints.html

File: templates/java/skeleton/templates/ApiQueries.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.beans.factory.annotation.Autowired;
+import lombok.extern.slf4j.Slf4j;
+
+import com.nx.${serviceName}.service.QueriesService;
+
+@RestController
+@RequestMapping("/apis/${serviceName}")
```
Comment (lsavouillannxw) [inline templates/java/skeleton/templates/ApiQueries.java]: api**S**?

File: templates/java/skeleton/templates/Application.java (hunk 4 above)
Comment (lsavouillannxw) [inline templates/java/skeleton/templates/Application.java]: Fixed

**Pull request merged**